src/flint/fmpz.jl

```diff
@@ -499,48 +499,28 @@ end
 
 function +(x::ZZRingElem, c::Int)
   z = ZZRingElem()
-  if c >= 0
-    ccall((:fmpz_add_ui, libflint), Nothing,
-          (Ref{ZZRingElem}, Ref{ZZRingElem}, Int), z, x, c)
-  else
-    ccall((:fmpz_sub_ui, libflint), Nothing,
-          (Ref{ZZRingElem}, Ref{ZZRingElem}, Int), z, x, -c)
-  end
+  add!(z, x, c)
   return z
 end
 
 +(c::Int, x::ZZRingElem) = x + c
 
 function -(x::ZZRingElem, c::Int)
   z = ZZRingElem()
-  if c >= 0
-    ccall((:fmpz_sub_ui, libflint), Nothing,
-          (Ref{ZZRingElem}, Ref{ZZRingElem}, Int), z, x, c)
-  else
-    ccall((:fmpz_add_ui, libflint), Nothing,
-          (Ref{ZZRingElem}, Ref{ZZRingElem}, Int), z, x, -c)
-  end
+  sub!(z, x, c)
```

Comment:
Note how this also replaces two `ccall` by just one (an even better implementation would also do inline arithmetic if `x` is a "small" integer, i.e. stored within the pointer)